tests/test_datasets/test_dataset.py

```diff
@@ -234,61 +212,48 @@ def setUp(self):
         self.sparse_dataset = openml.datasets.get_dataset(4136, download_data=False)
 
     def test_get_sparse_dataset_with_target(self):
-        X, y = self.sparse_dataset.get_data(
+        X, y, _, attribute_names = self.sparse_dataset.get_data(
             dataset_format='array', target="class"
         )
         self.assertTrue(sparse.issparse(X))
         self.assertEqual(X.dtype, np.float32)
         self.assertIsInstance(y, np.ndarray)
         self.assertIn(y.dtype, [np.int32, np.int64])
         self.assertEqual(X.shape, (600, 20000))
-        X, y, attribute_names = self.sparse_dataset.get_data(
-            dataset_format='array',
-            target="class",
-            return_attribute_names=True,
-        )
         self.assertTrue(sparse.issparse(X))
         self.assertEqual(len(attribute_names), 20000)
         self.assertNotIn("class", attribute_names)
         self.assertEqual(y.shape, (600, ))
 
     def test_get_sparse_dataset(self):
-        rval = self.sparse_dataset.get_data(dataset_format='array')
+        rval, _, categorical, attribute_names = self.sparse_dataset.get_data(dataset_format='array')
         self.assertTrue(sparse.issparse(rval))
         self.assertEqual(rval.dtype, np.float32)
         self.assertEqual((600, 20001), rval.shape)
-        rval, categorical = self.sparse_dataset.get_data(
-            dataset_format='array', return_categorical_indicator=True
-        )
-        self.assertTrue(sparse.issparse(rval))
+
         self.assertEqual(len(categorical), 20001)
         self.assertTrue(all([isinstance(cat, bool) for cat in categorical]))
-        rval, attribute_names = self.sparse_dataset.get_data(
-            dataset_format='array', return_attribute_names=True
-        )
-        self.assertTrue(sparse.issparse(rval))
+
         self.assertEqual(len(attribute_names), 20001)
-        self.assertTrue(all([isinstance(att, str)
-                             for att in attribute_names]))
+        self.assertTrue(all([isinstance(att, str) for att in attribute_names]))
 
     def test_get_sparse_dataframe(self):
-        rval = self.sparse_dataset.get_data(dataset_format='dataframe')
+        rval, *_ = self.sparse_dataset.get_data(dataset_format='dataframe')
```

Comment:
Isn't dataframe the new default? Could you adapt this for all unit test?

Reply:
It is. I am actually not entirely sure what is the best thing to do in unit tests; specify optional parameters or not.
I am actually leaning to specifying them if they are relevant to the test/expected output.

This way, you clearly create a unit test for each possible behavior explicitly. Whatever is then the default behavior should still work.

What do you think?

Reply:
Yes, that makes sense. Actually, it doesn't really work if you don't specifiy anything (as noted above), so this is totally fine.